igcollect/sfp_dom.py

```diff
@@ -0,0 +1,364 @@
+        # is similar to volts
+        self.metrics['tx_power']['current'] = a2_offset_to_volts(
+            self._data, SFF_A2_TX_PWR
+        )
+        self.metrics['tx_power']['low_warning'] = a2_offset_to_volts(
+            self._data, SFF_A2_TX_PWR_LWARN
+        )
+        self.metrics['tx_power']['high_warning'] = a2_offset_to_volts(
+            self._data, SFF_A2_TX_PWR_HWARN
+        )
+        self.metrics['tx_power']['low_critical'] = a2_offset_to_volts(
+            self._data, SFF_A2_TX_PWR_LALRM
+        )
+        self.metrics['tx_power']['high_critical'] = a2_offset_to_volts(
+            self._data, SFF_A2_TX_PWR_HALRM
+        )
+
+        # RX Power is 16-bit unsigned integer with LSB equal to 0.1uW
+        # We use the a2_offset_to_volts() function because the equivalence
+        # is similar to volts
+        self.metrics['rx_power']['current'] = a2_offset_to_volts(
+            self._data, SFF_A2_RX_PWR
+        )
+        self.metrics['rx_power']['low_warning'] = a2_offset_to_volts(
+            self._data, SFF_A2_RX_PWR_LWARN
+        )
+        self.metrics['rx_power']['high_warning'] = a2_offset_to_volts(
+            self._data, SFF_A2_RX_PWR_HWARN
+        )
+        self.metrics['rx_power']['low_critical'] = a2_offset_to_volts(
+            self._data, SFF_A2_RX_PWR_LALRM
+        )
+        self.metrics['rx_power']['high_critical'] = a2_offset_to_volts(
+            self._data, SFF_A2_RX_PWR_HALRM
+        )
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument('--prefix', default='sfp_dom')
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+    physical_interfaces = get_physical_interfaces()
+
+    for iface in physical_interfaces:
+        try:
+            data = ethtool_get_module_data(iface)
+            print_metrics(args.prefix, iface, data.metrics)
+        except (NoSFPException, UnsupportedSFPException) as exc:
+            print(exc, file=sys.stderr)
+            continue
+
+
+def get_physical_interfaces():
+    scn = '/sys/class/net'
+    physical_interfaces = []
+    for dev in listdir(scn):
+        dev_path = join(scn, dev)
+        if not (islink(dev_path) or isdir(dev_path)):
+            # Skip regular files
+            continue
+        # Physical interfaces have device as a symlink to /sys/class/device_id
+        if islink(join(scn, dev, 'device')):
+            physical_interfaces.append(dev)
+
+    return physical_interfaces
+
+
+def a2_offset_to_celsius(data, offset):
+    """
+    Returns a signed float built from a 16-bit signed twos complement value
+    in increments of 1/256 degrees Celsius
+    """
+    signed = ctypes.c_int16(a2_offset_to_int(data, offset))
+    return signed.value / 256
+
+
+def a2_offset_to_volts(data, offset):
+    return a2_offset_to_int(data, offset) / 10000
+
+
+def a2_offset_to_milliamps(data, offset):
+    return a2_offset_to_int(data, offset) / 256
```

Comment:
The conversion in line 292 is done by a decimal number 10k to convert a 100µV scale into V. 100µ * 10k = 1M and 1/1M is 1µ.  But here for the current which is given in 1µA units you divide by 256. Shouldn't you divide by 1k?

Reply:
I caught this mistake. 
The division here should have been 500, as each LSB is equivalent to 2µA.